ensembl-track-api.openapi.yaml

```diff
@@ -47,14 +47,33 @@ paths:
                   track_categories:
                     type: array
                     items:
-                      $ref: '#/components/schemas/TrackCategories'
+                      $ref: '#/components/schemas/TrackCategory'
         '400':
           description: Invalid request.
         '404':
           description: Specified genome ID was not found.
+  /tracks/{track_id}:
```

Comment:
Question: are we expecting track ids to be unique globally, or to only be unique within a given genome id? Currently, only tracks enabled via the expansion mechanism could have truly unique ids.

If track ids can't be globally unique, then the genome id should also be provided in the request. If so, how best to add it to the url?

Reply:
All track IDs currently are scoped to genome, so the URL should to include genome id (see [slack](https://genomes-ebi.slack.com/archives/CC2RKR525/p1692870200220029)).

Reply:
It is a question for the backend team how closely you want to follow the REST conventions.

- `/track_categories/:genome_id/:track_id`? (not conventional)
- `/track/:genome_id/:track_id`? (not conventional)
- `/tracks/:genome_id/:track_id`? (still not conventional)
- `/tracks/:track_id?genome_id=blah` (probably the most conventional, but might be uglier than the previous one)

Reply:
If we are going to use path parameters it should not be used to filter data but to identify a specific resource and it needs to be clear what is being identified.
I think the last example that @azangru provided is practical.
> `/tracks/:track_id?genome_id=blah`

Or if genome_id is always used to return a specific set of tracks it could be:
`/genome/:genome_id/track/:track_id`

Reply:
>  are we expecting track ids to be unique globally,

Does track_id ( the track being displayed) has any relevance without genome_id ?

Reply:
@kamaldodiya 

> Does track_id ( the track being displayed) has any relevance without genome_id ?

Yes and no.

Dan has used track ids to for triggers to genome browser.  Compare track ids reported by the track api [here](https://beta.ensembl.org/api/tracks/track_categories/a7335667-93e7-11ec-a39d-005056b38ce3) with triggers in the toml file ([here](https://github.com/Ensembl/ensembl-dauphin-style-compiler/blob/master/backend-server/config/boot-tracks/boot-tracks-16.toml#L12-L15)). He has been reusing the same track ids across different genomes.

However, with track expansion mechanism, you can have globally unique track ids that track api would be able to resolve. It, however, doesn't seem to be possible, in the api created by Dan, to provide a genome id alongside a track id.

Personally, I think globally unique identifiers are a much better decision. But, as you can see, there now is an incompatibility between the older mechanism of registering tracks (via boot-tracks.toml) and the newer one (via expansions).

😞  (but also 😠 😡 and 🤬)

Reply:
If we want to make an endpoint /track/:track_id (instead of /genome/:genome_id/track/:track_id), we need to make track IDs unique. But we can only do that for the tracks that have been migrated to expansion nodes (since eard files for static tracks rely on the current IDs) meaning that for most track IDs the endpoint will return multiple track payloads.

Reply:
Decisions made as a result of a meeting:

- Track ids will be unique, and will be UUIDs
- We are adding a `type` filed to the endpoint response. So far, the possible types are: `gene`, `variant`, and `regular`
- We will need to quickly confirm that the client can handle the unique track ids (see https://github.com/Ensembl/ensembl-client/blob/0945e691a4c00bf8d4b4713ba93cc37ce9de9c73/src/content/app/genome-browser/state/track-settings/trackSettingsSlice.ts#L189-L201)
- Going to add a file paths mapping to track api and inject those file paths into the variant eards. In the future, the file paths will be replaced with ids for the data file api.